mono/mini/mini-arm.c

```diff
@@ -7107,9 +7107,9 @@ mono_arch_build_imt_trampoline (MonoVTable *vtable, MonoDomain *domain, MonoIMTC
 		size += 4 * count; /* The ARM_ADD_REG_IMM to pop the stack */
 
 	if (fail_tramp)
-		code = mono_method_alloc_generic_virtual_trampoline (domain, size);
+		code = (arminstr_t*)mono_method_alloc_generic_virtual_trampoline (domain, size);
```

Comment:
This `arminstr_t`, that is otherwise usually `guint8` causes some grief in `ARM_EMIT`, that is solvable, but maybe change this to `guint8`, and then carefully downstream impact.